compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -9525,7 +9525,19 @@ private void handleSafeNavigation(BLangAccessExpression accessExpr, BType type,
     }
 
     private boolean isMapJson(BType originalType) {
-        return originalType.tag == TypeTags.MAP && ((BMapType) originalType).getConstraint().tag == TypeTags.JSON;
+        if (originalType.tag != TypeTags.MAP) {
+            return false;
+        }
+        BType constraintType = getConstraintType((BMapType) originalType);
+        return constraintType.tag == TypeTags.JSON;
```

Comment:
We can refactor this logic like the following.
```java
    private boolean isMapJson(BType originalType) {
        if (originalType.tag == TypeTags.MAP) {
            return isMapJson(((BMapType) originalType).getConstraint());
        }
        return originalType.tag == TypeTags.JSON;
    }
```

Reply:
This logic is going to return `true` when the `originalType` is `json`. Anyway we can accommodate that by calling `originalType.tag == TypeTags.MAP && isMapJson(originalType)`.

Reply:
or we can pass a boolean parameter like 
```java
 private boolean isMapJson(BType originalType, boolean fromMap) {
        if (originalType.tag == TypeTags.MAP) {
            return isMapJson(((BMapType) originalType).getConstraint(), true);
        }
        return originalType.tag == TypeTags.JSON && fromMap;
    }
```